giotto/mapper/visualization.py

```diff
@@ -54,238 +74,95 @@ def create_network_2d(graph, data, node_pos, node_color,
-                yanchor="top"
-            ),
-            go.layout.Updatemenu(
-                buttons=column_color_buttons,
-                direction="down",
-                pad={"r": 10, "t": 10},
-                showactive=True,
-                x=0.42,
-                xanchor='left',
-                y=button_height,
-                yanchor="top"
-            ),
-        ])
-
-    fig.update_layout(
-        annotations=[
-            go.layout.Annotation(text="Colorscale:", x=0, xref="paper",
-                                 y=button_height - 0.045, yref="paper",
-                                 align="left", showarrow=False)
-        ])
-
-    if columns_to_color is not None:
-        fig.add_annotation(
-            go.layout.Annotation(text="Color by:", x=0.37, xref="paper",
-                                 y=button_height - 0.045,
-                                 yref="paper", align="left", showarrow=False)
-
-        )
-
-    return fig
-
-
-def create_network_3d(graph, data, node_pos, node_color, columns_to_color=None,
-                      plotly_kwargs=None):
-    """
-    Parameters
+    References
     ----------
-    graph : igraph.Graph
-        The nerve of the refined pullback cover. Nodes correspond to cluster
-        sets, and an edge exists between two nodes if they share at least one
-        point in common.
+    .. [1] `igraph.Graph.layout
+            <https://igraph.org/python/doc/igraph.Graph-class.html#layout>`_.
 
-    data : ndarray of shape (n_samples, n_features)
-        The point cloud data used to generate the nerve.
+    """
 
-    node_pos : igraph.Graph.layout or ndarray of shape (n_nodes, n_dims)
-        Encodes the layout of the graph according to a layout algorithm or
-        pre-defined array of coordinates in an n-dimensional space.
+    # Compute the graph and fetch the indices of points in each node
+    pipe = clone(pipeline)
+    graph = pipe.fit_transform(data)
+    node_elements = graph['node_metadata']['node_elements']
 
-    node_color : ndarray of shape (n_nodes,)
-        The numerical values to color each node of the graph by.
+    # Simple duck typing to determine whether data is a pandas dataframe
+    is_data_dataframe = hasattr(data, 'columns')
 
-    columns_to_color : dict, optional, default: ``None``
-        Key-value pairs (column_name, column_index) to specify which columns of
-        :attr:`data` to color the graph by. Generates a dropdown widget to
-        select the columns to color by.
+    # Determine whether layout is an array of node positions
+    is_layout_ndarray = hasattr(layout, 'dtype')
+    if is_layout_ndarray:
+        node_pos = layout
+    else:
+        node_pos = graph.layout(layout, dim=layout_dim)
 
-    plotly_kwargs : dict, optional, default: ``None``
-        Keyword arguments to configure the Plotly Figure.
+    # Determine whether color_variable is an array or pandas series/dataframe
+    # containing scalar values
```

Comment:
Add (fit-)transformer and column index or list of indices in comment, or simply remove comment

Reply:
Removed since it's a bit redundant to have a docstring that says the same thing